inrupt/pod-browser
# Apps 1189 ensure adding contacts via web id works for all profile types

## Description
I began writing this PR with no small amount of confusion around how the new profiles affected PodBrowser. When I created this ticket, I didn't realize that this issue was unrelated to profiles. That being said, along the way I found some opportunities to clarify some things as I tend to make these kinds of small changes as I work through a story. This PR reflects those changes.

## Changes
- Updated the `addContact` component to take accommodate the changes
- Pulled `agentSearchForm` constants out of the component
- Changed the `value` prop on `agentSearchForm` for clarity
- Added a default empty array value for `permissions` on `agentSearchForm`
- Pulled the duplicate contact logic into a function
- Changed `filter` to `find` for checking existing contacts so `length` checking is unnecessary
- Created an `isPodOwner` utility because I saw this code duplicated in a couple places

## Testing
- Try adding a known working webID as a contact
- Verify you can add the contact

## Commit checklist


## Interested parties
@VirginiaBalseiro @solid-akb
